stdlib/2and3/_types.pyi

```diff
@@ -0,0 +1,7 @@
+from typing import Protocol, Union
+
+class _HasFileno(Protocol):
```

Comment:
Not sure it's necessary to give everything in this file a leading underscore. However, we should use the `@type_check_only` decorator.

Reply:
When I added `@type_check_only` I got a bunch of errors from pytype, any idea what caused them?
```
Ran pytype with 1754 pyis, got 30 errors.
stdlib/2/fcntl.pyi: ParseError: syntax error, unexpected CLASS, expecting DEF
stdlib/2and3/_types.pyi: ParseError: syntax error, unexpected CLASS, expecting DEF
stdlib/2and3/_types.pyi: ParseError: syntax error, unexpected CLASS, expecting DEF
stdlib/2and3/asynchat.pyi: ParseError: syntax error, unexpected CLASS, expecting DEF
...
```

Reply:
pytype's parser is a little on the strict side sometimes. @rchen152 might be able to help.

Reply:
There's a specific set of decorators that the pytype parser allows - looks like we never added `@type_check_only` to it. I'll go ahead and do that.

Reply:
Ok, I'll add it to HasFileno and mark this resolved.

Reply:
I've fixed the dev version of pytype - once our nightly tests run, I'll know whether we can do a pytype release tomorrow or if it'll take a few days. Either way, I'll update here.

Reply:
I pushed the new pytype version. I reran the pytype test, but it looks like it's still downloading the old version. I'll wait a bit and try the test again, and merge the PR if it passes.